Description: better fullscreen
 update fullscreen function, to go fullscreen and back, with F key
Author: Nicolas HOUDELOT <nicolas@demosdebs.org>
Applied-Upstream: no
Last-Update: 2024-03-28
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/main.c
+++ b/main.c
@@ -620,6 +620,13 @@
 	                      0,
 	                      (void (*)(void *, Uint8 *, int))play,
 	                      0};
+
+static void ToggleFullscreen(SDL_Window* sdlWindow) {
+    int FullscreenFlag = SDL_WINDOW_FULLSCREEN_DESKTOP;
+    int isFullscreen = SDL_GetWindowFlags(sdlWindow) & FullscreenFlag;
+    SDL_SetWindowFullscreen(sdlWindow, isFullscreen ? 0 : FullscreenFlag);
+    SDL_ShowCursor(isFullscreen);
+}
 #endif
 
 #if defined(WIN32) && !defined(SDL)
@@ -1106,6 +1113,11 @@
 end:
 			break;
 		}
+		if (event.type == SDL_KEYUP) {
+		    switch (event.key.keysym.sym) {
+		        case SDLK_f: ToggleFullscreen(sdlWindow); break;
+		    }
+    	}
 #endif
 #if defined(WIN32) && !defined(SDL)
 		glFlush();
@@ -1139,3 +1151,5 @@
 #endif
 #endif
 }
+
+
